spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -633,13 +672,21 @@ private Pair<String, List<String>> parseLocationString(String location) {
     }
   }
 
-  private Pair<Table, Long> loadFromPathIdentifier(PathIdentifier ident) {
+  @SuppressWarnings("CyclomaticComplexity")
+  private Table loadFromPathIdentifier(PathIdentifier ident) {
     Pair<String, List<String>> parsed = parseLocationString(ident.location());
 
     String metadataTableName = null;
     Long asOfTimestamp = null;
     Long snapshotId = null;
+    boolean isChangelog = false;
+
     for (String meta : parsed.second()) {
+      if (meta.equalsIgnoreCase(SparkChangelogTable.TABLE_NAME)) {
```

Comment:
`changelog` should be the last element of the list, right? this may have false match.

Reply:
This is for path-based tables, which have a bit weird identifiers like `location#meta1,meta2,meta3` so I am not sure whether changelog must be last. Let me think.

Reply:
I double checked this and I think we should follow the existing logic for path-based tables where the order of parts in a selector does not matter.